src/snippet/mod.rs

```diff
@@ -186,6 +186,29 @@ fn select_best_fragment_combination(fragments: &[FragmentCandidate], text: &str)
     }
 }
 
+/// Returns ranges that are collapsed into non-overlapped ranges.
+fn collapse_overlapped_ranges(ranges: &Vec<Range<usize>>) -> Vec<Range<usize>> {
+    let mut result = Vec::new();
```

Comment:
This assumes the ranges are sorted by start.
That's a hidden contract.

Please put that in the comment, and ideally check that with an assert.